src/main/java/com/cdancy/bitbucket/rest/features/UserApi.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.features;
+
+import com.cdancy.bitbucket.rest.annotations.Documentation;
+import com.cdancy.bitbucket.rest.domain.admin.UserPage;
+import com.cdancy.bitbucket.rest.fallbacks.BitbucketFallbacks;
+import com.cdancy.bitbucket.rest.filters.BitbucketAuthenticationFilter;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.RequestFilters;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Produces(MediaType.APPLICATION_JSON)
+@RequestFilters(BitbucketAuthenticationFilter.class)
+@Path("/rest/api/{jclouds.api-version}/user")
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public interface UserApi {
+
+    @Named("user:get-user")
+    @Documentation({"https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-rest.html#idp362"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/groups/more-members")
+    @Fallback(BitbucketFallbacks.UserPageOnError.class)
+    @GET
+    UserPage users(@Nullable @QueryParam("filter") String filter,
+                   @Nullable @QueryParam("group") String group,
+                   @Nullable @QueryParam("permission") String globalRole,
+                   @Nullable @QueryParam("permission.1") String projectRole,
+                   @Nullable @QueryParam("permission.1.projectKey") String projProjectKey,
+                   @Nullable @QueryParam("permission.1.projectId") String projProjectId,
+                   @Nullable @QueryParam("permission.2") String repoRole,
+                   @Nullable @QueryParam("permission.2.projectKey") String repoProjectKey,
+                   @Nullable @QueryParam("permission.3.projectId") String repoPrrojectId,
```

Comment:
So we could have N number of these which is going to make this a bit unruly and also limit folks if we only specify 3. We'll probably have to create an optional Binder which will take in a list of of these "permission objects" which we will then have to inject the values as query-params. Take a look [HERE](https://github.com/cdancy/bitbucket-rest/blob/master/src/main/java/com/cdancy/bitbucket/rest/binders/BindHookSettingsToPayload.java) and poke around the code base to see how we do this type of thing elsewhere. Let me know if you have any questions. Looks good so far.

Reply:
I'm sorry I misunderstood the documentation I thought permission was for global 1 was for project permissions and 2 for repo etc.. I didn't recognize they were the same thing but different filters.

I looked into what you told me, isn't that a way to bind some data to the payload? which is as much as I know should not be used with GET requests as their body is not processed? Please correct me if I misunderstood anything.

Reply:
Ignore the class name ... you can alter the request in general and append new query params/etc. Take a look at the below code where we do something like this:

https://github.com/cdancy/jenkins-rest/blob/master/src/main/java/com/cdancy/jenkins/rest/filters/ScrubNullFolderParam.java

Reply:
I am sorry I have to postpone it to the next Friday, as am having my midterm exams this week and I got a lot of studying to do.

Reply:
@ajammil of course and take your time.